File: 알고리즘 정리/그리디 알고리즘/문제풀이/큰 수 만들기.py
def solution(number, k):
    answer = ''
    idx=0
    n=len(number)-k
    while n>0:
        max_num='0'
        for i in range(idx,len(number)-n+1):
            if max_num<number[i]:
                max_num=number[i]
                idx=i
            if number[i]=='9':
                break
        idx+=1
        answer+=max_num
        n-=1
    return answer

File: 알고리즘 정리/그리디 알고리즘/문제풀이/큰 수 만들기.cpp
#include <string>
#include <vector>
using namespace std;

string solution(string number, int k) {
	string answer = "";
	int pos = 0;
	int len = number.size() - k;
	while (k > 0 && answer.size() < len)
	{
		char max_num = '0';
		int cnt = 0;
		for (int i = pos; i <= pos + k; i++)
		{
			if (max_num < number[i])
			{
				max_num = number[i];
				cnt = i - pos;
			}
		}
		pos += cnt + 1;
		answer += number[pos - 1];
		k -= cnt;
	}
	if (k == 0)
		for (int i = pos; i < number.size(); i++)
			answer += number[i];
	return answer;
}